Contest/Assignment/src/org/togetherjava/event/elevator/elevators/Elevator.java

```diff
@@ -76,15 +91,91 @@ public void moveOneFloor() {
         //  meaning that the average time waiting (either in corridor or inside the elevator)
         //  is minimized across all humans.
         //  It is essential that this method updates the currentFloor field accordingly.
+        if(requestedDestinationFloors.isEmpty()) {
+            lastResort();
+            return;
+        }
+        sortingList();
+        int currentRequest = requestedDestinationFloors.getLast();
+        if(currentRequest > currentFloor) {
+            currentFloor++;
+            travellingDirection = TravelDirection.UP;
+        }
+        else if(currentRequest < currentFloor) {
+            currentFloor--;
+            travellingDirection = TravelDirection.DOWN;
+        }
+        else {
+            requestedDestinationFloors.removeLast();
+            removeRequests();
+        }
         System.out.println("Request to move a floor received");
     }
 
+    /**
+     * It is possible that the requests ends for the current elevator but there are still humans onboard.
+     * In that case the paternoster is the last resort to make sure the humans onboard reach the destination floor.
+     */
+    private void lastResort() {
+        if(currentFloor == minFloor) travellingDirection = TravelDirection.UP;
+        else if(currentFloor == (minFloor + floorsServed -1)) travellingDirection = TravelDirection.DOWN;
+        else if(numberOfHumansAboard != 0) {
+            if(travellingDirection == TravelDirection.UP) currentFloor++;
+            else if(travellingDirection == TravelDirection.DOWN) currentFloor--;
+        }
+    }
+
+    /**
+     * For instance, a number of humans can exit to the current floor.
+     * But the request for the current floor can make the elevator come to the current floor again or stay at it for consecutive steps.
+     * To prevent such loss of steps, this method removes the number of request according to the number of humans exited.
+     */
+    private void removeRequests() {
+        for(int i = 0; i < humanExited; i++) {
+            for(int j = 0; j < requestedDestinationFloors.size(); j++) {
+                if(requestedDestinationFloors.get(j) == currentFloor) {
+                    requestedDestinationFloors.remove(j);
+                    humanExited--;
+                    j--;
+                }
+            }
+        }
```

Comment:
that logic looks a bit wobbly. especially with that outer loop, `i` isnt used anywhere either. smells a bit.